clarity.d.ts

```diff
@@ -73,12 +73,16 @@ interface IPayload {
 }
 
 interface IEnvelope {
+  impressionId: string;
+  time?: number;
+  sequenceNumber?: number;
+}
+
+interface IClientInfo {
```

Comment:
Have you considered keeping all this info in envelope except not repeating redundant stuff in subsequent envelopes to save on bytes. This will ensure that if for some reason top URL changes (like # appended) we will still capture it but won't waste bytes if nothing changed?

Reply:
I have, but i thought that it wasn't clear enough. If we did that, we would be relying on previous payloads constantly and would have to re-scan entire data from start to end to see what was going on. Everything, except in rare cases url, is constant in this object. If we think it's important to track url changes, we should probably fire a separtate 'Redirect' event.

---
In reply to: [144364338](https://github.com/Microsoft/clarity-js/pull/39#discussion_r144364338) [](ancestors = 144364338)

Reply:
Synced offline: We renamed this to 'metadata' and we'd be passing it as a part of the first payload.

First payload will have the following structure:
{
  envelope: IEnvelope;
  events: IEvent[];
  metadata: IImpressionMetadata;
}

Subsequent payloads will just contain envelope and events.

---
In reply to: [144372367](https://github.com/Microsoft/clarity-js/pull/39#discussion_r144372367) [](ancestors = 144372367,144364338)